### DISCUSSION TITLE
RA_outpu error

### DISCUSSION THREAD

**Discussion by Shivani8898** [egeulgen/pathfindR#149] [Q&A]
Hi,
I am using PathfindR for pathway enrichrichment(KEGG).
No wi am facing problem from last 2 weeks earler it was fine but i don't understand what is the problem now.
one time it is running properly another time no RA_output.
Can you please help me with that.
I tried everthing from shutdown to reinstalling R. Also, the default version of PathfindR is showing error.
This is what I get:

There is already a directory named "pathfindR_Results".
Writing the result to "pathfindR_Results(18)" not to overwrite any previous results.
## Testing input
The input looks OK
## Processing input. Converting gene symbols,
          if necessary (and if human gene symbols provided)
Number of genes provided in input: 9066
Number of genes in input after p-value filtering: 9066
pathfindR cannot handle p values < 1e-13. These were changed to 1e-13
Could not find any interactions for 1061 (11.7%) genes in the PIN
Final number of genes in input: 8005
## Annotating involved genes and visualizing enriched terms
Downloading pathway diagrams of 236 KEGG pathways

  |===================                                                                                     |  18%The file /var/folders/k3/2bypsbdn7mgb7n3gt5q6jhhc0000gn/T//RtmpXI6Jhu/file33745a3de0 seems not to be a valid KGML file

Detailed error messages fromXML::xmlTreeParse:
Error: 1: Couldn't find end of Start Tag rela line 1374
2: Premature end of data in tag pathway line 4
> View(RA_output6)
Error in View : object 'RA_output6' not found

**Comment by egeulgen** [comment]
This seems to be an issue with obtaining the KEGG pathways? Which version of pathfindR are you using? I suspect updating to the latest version (along with updating the dependencies) should resolve the issue as there were issues with KEGGREST before

**Comment by Shivani8898** [comment]
Hi, actually i am using the latest version(1.6.4)

**Reply by egeulgen** [reply]
what version of KEGGREST are you using?

**Comment by Shivani8898** [comment]
kEGGREST VERSON-1.36.3

**Reply by egeulgen** [reply]
as this seems to be an issue with the visualization KEGG pathways, I will suggest running without visualization and performing visualization separately:

```r
out_df <- run_pathfindR(in_df, visualize_enriched_terms = FALSE)
visualize_terms(out_df, input_processing(in_df))
```

The KGML file (basically XML) seems to be corrupt for some reason. From the percentage (18%), the problematic term seems to be number 42 (236 * 0.18), I would specifically check that: 

```r
visualize_terms(out_df[42, ], input_processing(in_df))
```

However, I suspect that this might be due to a connection issue or (more likely) there might be a request limitation for the KEGG REST API that I'm not aware of.

In summary, I'd suggest running `run_pathfindR()` with `visualize_enriched_terms = FALSE` to obtain the actual enrichment results and try to debug the visualization issue separately

**Comment by Shivani8898** [comment]
Thank you so much, this worked.

**Reply by Shivani8898** [reply]
Hello, 
Thanks for your help in resolving the error.
I have another question related to the term _gene_ graph. I wanted to know how can I decrease the node size so that all the terms can be fitted.
I used the following codes and got this warning message.
term_gene_graph(result_df = RA_output1, use_description = TRUE)
1: ggrepel: 1 unlabeled data points (too many overlaps). Consider increasing max.overlaps 
2: ggrepel: 1 unlabeled data points (too many overlaps). Consider increasing max.overlaps 
3: ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps 
4: ggrepel: 3 unlabeled data points (too many overlaps). Consider increasing max.overlaps

**Comment by egeulgen** [comment]
Hello,

You can change the node sizes as follows:

```r
g <- term_gene_graph(RA_output)
g <- g + ggplot2::scale_size(range = c(2, 5), name = "# genes") # depending on your choice (when node_size == "num_genes")
g <- g + ggplot2::scale_size(range = c(2, 5), name = "-log10(p)") # depending on your choice (when node_size == "p_val")
g
```
